### ISSUE TITLE
Impossible to download image with status code of 404

### ISSUE THREAD

**Issue by rickharrison** [SDWebImage/SDWebImage#3226]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | iOS
 Platform Version        | 14.5
 SDWebImage Version      | 5.10.0
 Integration Method      | Carthage
 Xcode Version           | Xcode 12.5
 Repro rate              | 100%
 Repro with our demo prj | 
 Demo project link       | 

### Issue Description and Steps

Even when a server responds with image data, it seems it is impossible to download the image data. Would it be possible to download an image if the data is there despite the status code? Here is an example of the headers for a response which responds with an error using SDWebImage:

```
HTTP/2 404 
content-type: image/png
accept-ranges: bytes
date: Wed, 05 May 2021 21:11:27 GMT
via: 1.1 varnish
server: snooserv
content-length: 1048
```

**Comment by dreampiggy** [comment]
Seems...You want to custom the `HTTP Response Status Code` check logic ?


This sounds a feature request, similiar to Kingfisher's [isValidStatusCode API](https://github.com/onevcat/Kingfisher/blob/6faf189e7e2bde16cc50c8b28eb8245b00822398/Sources/Networking/ImageDownloaderDelegate.swift#L118)

**Comment by dreampiggy** [comment]
Can be easily achieve actually.

For current SDWebImage version, can you try using the [ResponseModifier](https://sdwebimage.github.io/SDWebImage/Classes/SDWebImageDownloaderResponseModifier.html) ?

This powerful feature allows you to even mock the server HTTP response (including status code). So it's simple that you can always return 200 instead of what server get.

**Comment by dreampiggy** [comment]
@rickharrison
See #3227. For your case, you can config `acceptableStatusCodes` to `nil`, then all the status code will passthrough without failure.

Which means, we only use the image data decoding to detect whether this download is success or not.

**Comment by rickharrison** [comment]
Yes, that looks perfect! This will really help when you are not in control of the servers that are returning images with an odd status code. Thank you!!!
